WordPress/gutenberg
# Palette Edit: Don't discards colors with default name and slug

Fixes: #37861
Related to #37496

## What?
This PR will not discard colors with the default name (`Color-{number}`) and default color (`#000`) in the `PaletteEdit` component.

## Why?

I have looked at the changes made in #37496 and this PR makes two main changes:

- Apply the default slug when adding a new color.
- If a color has a default slug and a default color, discard that color as _temporary_

However, my feeling is that the newly added color will have an appropriate name and slug so that the color no longer needs to be considered a "temporary" color.

In addition, you can never save a color that is considered a temporary color (i.e., one whose name is in the form `color {number}` and whose HEX color is `#000`). To me, this behavior feels more like a bug.

## How?

I removed the code related to the `isTemporaryElement()` function that determines if a color is temporary or not, so that colors with default name are not deleted. I also adjusted variable names and comments.

Note: The `PaletteEdit` component generates a slug from the input string, but problems exist regarding non-Latin and multibyte characters. See #49711 and #39210 for details.

## Testing Instructions

- Access the Global Style panel.
- Press the "Add" button on the Custom Color palette in succession.
- Colors named `Color 1`, `Color 2`, and `Color 3` should be created.
- Click the "Done" button.
- The color palette should be retained.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/54422211/6d956b8a-8724-4d43-a0c5-87d8bb4d09e9
